neural_modelling/src/neuron/spike_processing.c

```diff
@@ -257,32 +289,46 @@ static inline void setup_synaptic_dma_write(
 
 //! \brief Called when a multicast packet is received
 //! \param[in] key: The key of the packet. The spike.
-//! \param payload: Ignored
-static void multicast_packet_received_callback(uint key, UNUSED uint payload) {
-    log_debug("Received spike %x at %d, DMA Busy = %d", key, time, dma_busy);
-
-    // If there was space to add spike to incoming spike queue
-    if (in_spikes_add_spike(key)) {
-        // If we're not already processing synaptic DMAs,
-        // flag pipeline as busy and trigger a feed event
-        // NOTE: locking is not used here because this is assumed to be FIQ
-        if (!dma_busy) {
-            log_debug("Sending user event for new spike");
-            if (spin1_trigger_user_event(0, 0)) {
-                dma_busy = true;
-            } else {
-                log_debug("Could not trigger user event\n");
-            }
-        }
+//! \param payload: the payload of the packet. The count.
+static void multicast_packet_received_callback(uint key, uint payload) {
+    p_per_ts_struct.packets_this_time_step += 1;
+
+    // handle the 2 cases separately
+    if (payload == 0) {
+        log_debug(
+            "Received spike %x at %d, DMA Busy = %d", key, time, dma_busy);
+        // set to 1 to work with the loop.
+        payload = 1;
     } else {
-        log_debug("Could not add spike");
+        log_debug(
+            "Received spike %x with payload %d at %d, DMA Busy = %d",
+            key, payload, time, dma_busy);
+    }
+
+    // cycle through the packet insertion
+    for (uint count = payload; count > 0; count--) {
+        in_spikes_add_spike(key);
+    }
+
+    // If we're not already processing synaptic DMAs,
+    // flag pipeline as busy and trigger a feed event
+    // NOTE: locking is not used here because this is assumed to be FIQ
+    if (!dma_busy) {
+        log_debug("Sending user event for new spike");
+        if (spin1_trigger_user_event(0, 0)) {
+            dma_busy = true;
+        } else {
+            log_warning("Could not trigger user event\n");
+        }
     }
 }
 
 //! \brief Called when a DMA completes
 //! \param unused: unused
 //! \param[in] tag: What sort of DMA has finished?
 static void dma_complete_callback(UNUSED uint unused, uint tag) {
+    use(unused);
```

Comment:
Not critical but not needed

Reply:
same question. has something changed that allows us to not have that?

Reply:
Note the UNUSED in the line above

Reply:
fixed